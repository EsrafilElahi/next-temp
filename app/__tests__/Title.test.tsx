import { cleanup, fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Title from "@/app/_components/Title";

afterEach(() => {
  cleanup();
});

beforeEach(() => {
  render(<Title handleScroll={jest.fn()} />);
});

describe("Title Component", () => {
  test("2 title and 1 btn exists in document", () => {
    const nameTitle = screen.getByRole("heading", { name: /Esrafil Elahi/i });
    const positionTitle = screen.getByRole("heading", {
      name: /Senior Frontend Engineer/i,
    });
    const btnAboutMe = screen.getByRole("button");

    expect(nameTitle).toBeInTheDocument();
    expect(positionTitle).toBeInTheDocument();
    expect(btnAboutMe).toBeInTheDocument();
  });

  test("snapshot test", () => {
    const { container } = render(<Title handleScroll={jest.fn()} />);
    expect(container).toMatchSnapshot();
  });
});
