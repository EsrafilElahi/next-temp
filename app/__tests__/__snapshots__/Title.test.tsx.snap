// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title Component snapshot test 1`] = `
<div>
  <div
    class="absolute top-[45%] left-[5%]"
  >
    <h1
      class="text-[7vmin] md:text-[10vmin] leading-[2.5rem] md:leading-[5rem]"
    >
      Esrafil Elahi
    </h1>
    <h2
      class="text-[3.5vmin] md:text-[5vmin] mb-3 md:mb-5"
    >
      Senior Frontend Engineer
    </h2>
    <button
      class="btn-outline xs-only:text-[3vmin]"
    >
      About Me
    </button>
  </div>
</div>
`;
